### PR TITLE
feature: update `useDappAudience` to take into account address types

### PR DESCRIPTION
**Pull request by Kiryous**

### CODE REVIEW

File: packages/dialect-react-ui/hooks/useDappAudience.ts [hunk 1]
```diff
@@ -28,22 +30,31 @@ interface UseDappAudienceValue {
...
   } = useDappAddresses({
     refreshInterval,
   });
   const isFetching = isFetchingSubscriptions || isFetchingAddresses;
 
+  if (addressTypes) {
+    addresses = addresses.filter((addr) =>
+      addressTypes.includes(addr.address.type)
+    );
+  }
```
Comment (zhelezkov) [inline packages/dialect-react-ui/hooks/useDappAudience.ts:56]: looks soo mutable, can we change it to immutable expression? Afraid that someone might get struggled while debugging something here

File: packages/dialect-react-ui/hooks/useDappAudience.ts [hunk 2]
```diff
@@ -28,22 +30,31 @@ interface UseDappAudienceValue {
...
   notificationTypeId,
+  addressTypes,
   refreshInterval = DEFAULT_ADDRESSES_REFRESH_INTERVAL,
 }: UseDappAudienceParams): UseDappAudienceValue {
   const {
     subscriptions: notificationsSubscriptions,
     isFetching: isFetchingSubscriptions,
     errorFetching: errorFetchingNotificationSubscriptions = null,
   } = useDappNotificationSubscriptions();
+  let { addresses } = useDappAddresses({
```
Comment (zhelezkov) [inline packages/dialect-react-ui/hooks/useDappAudience.ts:41]: let's change it to const

**Review by zhelezkov — CHANGES_REQUESTED** [review]

File: packages/dialect-react-ui/hooks/useDappAudience.ts (hunk 1 above)
Comment (Kiryous) [reply packages/dialect-react-ui/hooks/useDappAudience.ts]: hey @zhelezkov, changed this part a bit, wdyt?

**Review by zhelezkov — APPROVED** [review]
much better!

**Pull request merged**